libsql-server/src/hrana/ws/session.rs

```diff
@@ -72,11 +74,21 @@ pub(super) async fn handle_initial_hello<F: MakeNamespace>(
 ) -> Result<Session<<F::Database as Database>::Connection>> {
     let namespace_jwt_key = server
         .namespaces
-        .with(namespace, |ns| ns.jwt_key())
+        .with(namespace.clone(), |ns| ns.jwt_key())
         .await??;
+
+    // Convert jwt token into a HeaderValue to be compatible with UserAuthStrategy
+    let user_credential = jwt
+        .clone()
+        .and_then(|t| HeaderValue::from_str(&format!("Bearer {t}")).ok());
+
     let authenticated = server
-        .auth
-        .authenticate_jwt(jwt.as_deref(), server.disable_namespaces, namespace_jwt_key)
+        .user_auth_strategy
+        .authenticate(UserAuthContext {
+            namespace,
+            user_credential,
+            namespace_credential: namespace_jwt_key,
+        })
```

Comment:
What are the thoughts on maybe having this function accept these each as individual parameters rather than a struct? I feel like this could reduce some of the noise here.

Reply:
I think having a struct gives more flexibility of the shape of `UserAuthContext` long term with less impact on strategy implementations if it did unless we removed fields. I am open to moving these to parameters.

Reply:
Okay I see, it would break changes to the consumer of this struct (which is just internal so that is okay) but wouldn't be a breaking change (to add new fields) to implementers which could exist outside of this code base. This makes sense to me!